### PR TITLE
fix: azure interactive caching

### PR DESCRIPTION
**Pull request by jon-stewart**
## Summary

CLI command, generate Azure TF, wasn't making use of the values loaded from cache.

## How did you test this change?

- unit testing
- integration testing
- manual testing

## Issue

[ALLY-1217](https://lacework.atlassian.net/browse/ALLY-1217)

### CODE REVIEW

**Review by rmoles — APPROVED** [review]
Looks good to me

File: cli/cmd/generate_azure.go [hunk 1]
```diff
@@ -317,6 +317,7 @@ By default, this command will function interactively, prompting for the required
 )
 
 type AzureGenerateCommandExtraState struct {
+	AskAdvanced    bool
```
Comment (ipcrm) [inline cli/cmd/generate_azure.go:320]: Looks like this is done inconsistently (before and now).  GCP/Azure (with your change) both cache this, whereas AWS does not. I think it makes sense to cache this. If the customer answered advanced questions previously and is resuming the cache, probably makes sense to re-prompt them with everything they may have selected before.

Not sure if we should raise a different PR for that to not muddy the water here; but figured it was worth calling out.

**Review by ipcrm — APPROVED** [review]
Looks good! One question

File: cli/cmd/generate_azure.go (hunk 1 above)
Comment (jon-stewart) [reply cli/cmd/generate_azure.go:320]: Good spot, I agree as we cache each of the advanced options.  I'll include in this PR.

**Pull request merged**